lib/application.ts

```diff
@@ -910,18 +910,43 @@ export class SNApplication {
   }
 
   /**
-   * Creates a JSON string representing the backup format of all items, or just subItems
-   * if supplied.
+   * @returns a JSON string representing the encrypted backup format of all
+   * items, including keyParams and protocol version.
    */
-  public async createBackupFile(
-    subItems?: SNItem[],
-    intent?: EncryptionIntent,
-    returnIfEmpty = false
-  ) {
+  public async createEncryptedBackupString(): Promise<string> {
+    const backup = await this.protocolService!.createBackupFile(
+      EncryptionIntent.FileEncrypted
+    );
+    return JSON.stringify(backup, null, 2);
+  }
+
+  /**
+   * @returns a JSON string representing the decrypted backup format of all
+   * items, including keyParams and protocol version.
+   */
+  public async createDecryptedBackupString(): Promise<string | undefined> {
+    const backup = await this.createDecryptedBackup();
+    if (backup) {
+      return JSON.stringify(backup, null, 2);
+    } else {
+      return undefined;
+    }
+  }
+
+  public async createDecryptedBackup(): Promise<BackupFile | undefined> {
+    if (
+      this.hasPasscode() &&
```

Comment:
If no passcode, we'd still want to prompt for account password if account.

Reply:
Do we want to prompt for both? If not then we need to require authentication for `removePasscode()` as well, otherwise it becomes easy to circumvent.

Reply:
I decided to make it prompt for both for safety.

Reply:
Yeah, I guess it doesn't hurt. If you're bulk ejecting data we want to make it as safe as possible.

Reply:
@baptiste-grob the only downside is that some people download a decrypted backup when they have forgotten their account password..